### PR TITLE
Searchable dropdowns

### PR DESCRIPTION
**Pull request by rgantzosonscratch**
A simple way to search dropdowns in the code editor. You can search through selection boxes such as sprites to clone, costumes to display, and more!

### CODE REVIEW

**Comment by rgantzosonscratch** [comment]
I'll merge this, but I just want to check the compatibility with the Firefox version's repo first.

**Comment by RedGuy12** [comment]
i thought you didnt add things that scratch addons has?

**Comment by rgantzosonscratch** [comment]
> https://github.com/ScratchAddons/ScratchAddons/tree/master/addons/editor-searchable-dropdowns
> image

Well this is already in our Firefox version, and any Firefox users can see it.

File: features/features.json [hunk 1]
```diff
@@ -33,6 +33,13 @@
 		"urls": ["https://scratch.mit.edu/users/rgantzos/"],
 		"file": "advanced-search"
 	},
+	{
+		"title":"Searchable Dropdowns",
+		"description":"Easily search through options in the code editor. You can search for variables, costumes, and more!",
+	 	"credits":["Garbomuffin"],
```
Comment (GarboMuffin) [inline features/features.json:39]: if you're going to put my name here, at least format it correctly
```suggestion
	 	"credits":["GarboMuffin"],
```

File: features/searchable-dropdown.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+// ==UserScript==
```
Comment (GarboMuffin) [inline features/searchable-dropdown.js:1]: Reminder that this script is over three years old (https://github.com/forkphorus/cat-plus/blob/master/searchable-dropdowns.user.js) and contains a massive amount of bugs. The equivalent addon in Scratch Addons (which was also originally based on this script) has received many significant fixes and features that are not present here.

**Review by GarboMuffin — COMMENTED** [review]
Some comments you may want to keep in mind

File: features/searchable-dropdown.js (hunk 2 above)
Comment (rgantzosonscratch) [reply features/searchable-dropdown.js:1]: Alright, thank you very much! I will ask the beta testers to look out for bugs (which so far they haven't found any).

Comment (GarboMuffin) [reply features/searchable-dropdown.js:1]: Here's a few from memory to get you started:

 - can't press enter to select an item
 - arrow keys don't work properly
 - items are not sorted in any way
 - hovering over an item steals focus from the search bar
 - if I remember correctly, it interferes with angle inputs and maybe color ones too
 - the size of the dropdown can change as you search
 - ...

(The Scratch Addons version does not have any of these issues)

**Comment by RedGuy12** [comment]
why are you still working on this, just remove it from FF next update and everything’ll be fine

**Comment by rgantzosonscratch** [comment]
> why are you still working on this, just remove it from FF next update and everything’ll be fine

Why? We’re working on it so that we can add it to our next Chrome release.

**Comment by RedGuy12** [comment]
> Why?

Because Scratch Addons has this?

How it's looking to me is that you don't add things SA already has *unless you already made/ported it before checking*...

**Comment by rgantzosonscratch** [comment]
> > Why?
> 
> Because Scratch Addons has this?
> 
> How it's looking to me is that you don't add things SA already has _unless you already made/ported it before checking_...

Here’s how it works for us: if there’s an original feature that another extension has, then we don’t take it. However, this came from a separate userscript that another extension simply borrowed. We are doing the same.

**Comment by RedGuy12** [comment]
a  very buggy userscript that the creator of discourages use of

**Comment by rgantzosonscratch** [comment]
> a very buggy userscript that the creator of discourages use of

GarboMuffin does not discourage the use of it- he discourages the use of the current version because it has bugs. That’s why it’s still WIP.

**Pull request closed without merging**